### PR TITLE
Feat/product state

### PR DESCRIPTION
**Pull request by jgfidelis**
https://fidelis--storecomponents.myvtex.com/classic-shoes/p
https://fidelis--samsungar.myvtex.com/galaxy-s10/p
https://fidelis--alssports.myvtex.com/sitka-hoody-kelvin-lite-optifade/p

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+
+- Removed ProductContextProvider and left only on ProductWrapper.
```
Comment (kevinch) [inline CHANGELOG.md]: Changed or Removed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+
+### Added
+
+- Reducer and state to be changed by children components of product page.
```
Comment (kevinch) [inline CHANGELOG.md]: Sounds like a changed rather than an addition.

File: react/ProductWrapper.js [hunk 3]
```diff
@@ -98,22 +196,7 @@ const ProductWrapper = ({
...
-    () => ({
-      product,
-      categories: path(['categories'], product),
-      selectedItem,
-      onChangeQuantity: setSelectedQuantity,
-      selectedQuantity,
-    }),
-    [product, selectedItem, setSelectedQuantity, selectedQuantity]
-  )
+  const dispatchValue = useMemo(() => ({ dispatch }), [dispatch])
```
Comment (klzns) [inline react/ProductWrapper.js]: No need to useMemo

File: react/ProductWrapper.js [hunk 4]
```diff
@@ -18,8 +104,20 @@ const ProductWrapper = ({
   ...props
 }) => {
   const { account } = useRuntime()
+  const items = path(['items'], product) || []
 
-  const [selectedQuantity, setSelectedQuantity] = useState(1)
+  const [state, dispatch] = useReducer(reducer, {
```
Comment (klzns) [inline react/ProductWrapper.js]: Lazy initialize

**Pull request merged**